crates/hdk3/src/capability.rs

```diff
@@ -1,5 +1,199 @@
-pub mod create_cap_claim;
-pub mod create_cap_grant;
-pub mod delete_cap_grant;
-pub mod generate_cap_secret;
-pub mod update_cap_grant;
+use crate::prelude::*;
+
+/// Create capability claims on the local source chain.
+///
+/// Wraps the `create` HDK3 function with system type parameters set.
+/// This guards against sending application entry data or setting the wrong entry type.
+///
+/// Capability claims are grant _secrets_ that have been received from someone else.
+/// The grant entry is never sent, only the associated secret.
+/// The claim should be created on the local source chain so that it can be retrieved for later use.
+///
+/// Grantees of CapGrant secrets use CapClaim entries to save the secret.
+///
+/// The `CapClaim` contains the secret, tag and issuing agent.
+/// Only the secret should ever be sent back to the issuing agent, everything else is only for local
+/// administering, querying and filtering.
+///
+/// There is no guarantee that any CapClaim is currently usable without trying it.
+/// The author of the corresponding CapGrant can revoke it at any time or be unreachable on the network.
+///
+/// Any `call_remote` will return a `ZomeCallResponse::Unauthorized` when the grantor considers the
+/// secret invalid for the call. The caller is expected to handle this gracefully.
+///
+/// If the author of the CapGrant is reachable on the network and has not revoked the grant they will allow any
+/// agent with a valid secret and pubkey to `call_remote` externs on the grant author's machine.
+///
+/// Functions executed on the grant author's machine that change the grantor's source chain will be signed by
+/// the grantor.
+/// Delegating responsibility to grantee claimants is a serious responsibility!
+///
+/// @see CapClaim
+/// @see other cap grant functions
+pub fn create_cap_claim(cap_claim_entry: CapClaimEntry) -> ExternResult<HeaderHash> {
+    create(EntryWithDefId::new(
+        EntryDefId::CapClaim,
+        Entry::CapClaim(cap_claim_entry),
+    ))
+}
+
+/// Create a capability grant.
+///
+/// Wraps the `create` HDK3 function with system type parameters set.
+/// This guards against sending application entry data or setting the wrong entry type.
+///
+/// Capability grants are explicit entries in the local source chain that grant access to functions running in the current conductor.
+/// The grant must be sent (e.g. with a remote call) to the grantees so they can commit a claim and then call back with it in the future.
+///
+/// When an agent wants to expose externs to be called remotely by other agents they need to select
```

Comment:
I wonder if we can call them "zome functions" at this level of detail? Or maybe we should stop talking about zome functions and call them "externs"? or something else? "Extern" just sounds very low-level FFI, something the average dev might not relate to.

```suggestion
/// When an agent wants to expose zome functions to be called remotely by other agents they need to select
```